src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -90,53 +142,112 @@ export default class CoExpressionTab extends React.Component<ICoExpressionTabPro
+                    const profiledSampleCount = this.props.store.molecularProfileIdToProfiledSampleCount.result![profile.molecularProfileId];
+                    return {
+                        label: `${profile.name} (${profiledSampleCount} sample${profiledSampleCount !== 1 ? "s" : ""})`,
+                        value: profile.molecularProfileId
+                    };
+                });
             return (
-                <div
+                <div>
+                    <div
+                        style = {{
+                            display: "flex",
+                            alignItems: "center",
+                            padding: "5px"
+                        }}
+                    >
+                        <span>Subject Profile:</span>
+                        <div style={{display:"inline-block", width:376, marginLeft:4, marginRight:4, zIndex:15 /* so that on top when opened*/}}>
+                            <Select
+                                name="subject-profile-select"
+                                value={this.isSelectedGeneticEntityAGeneSet ? this.selectedSubjectProfileGeneset.molecularProfileId : this.selectedSubjectProfileGene.molecularProfileId}
+                                onChange={this.isSelectedGeneticEntityAGeneSet ? this.onSelectSubjectProfileGeneset : this.onSelectSubjectProfileGene}
+                                options={this.isSelectedGeneticEntityAGeneSet ? subjectOptionsGeneset : subjectOptionsGene}
+                                searchable={false}
+                                clearable={false}
+                                disabled={this.isSelectedGeneticEntityAGeneSet ? true : false}
+                                className="coexpression-select-subject-profile"
+                            />
+                        </div>
+                    </div>
+                    <div
                     style = {{
                         display: "flex",
-                        alignItems: "center"
+                        alignItems: "center",
+                        padding: "5px"
                     }}
-                >
-                    <span>Profile:</span>
-                    <div style={{display:"inline-block", width:376, marginLeft:4, marginRight:4, zIndex:10 /* so that on top when opened*/}}>
-                        <MolecularProfileSelector
-                            name="data-set-select"
-                            value={this.selectedMolecularProfile.molecularProfileId}
-                            onChange={this.onSelectDataSet}
-                            molecularProfiles={this.props.store.coexpressionTabMolecularProfiles.result}
-                            molecularProfileIdToProfiledSampleCount={this.props.store.molecularProfileIdToProfiledSampleCount.result}
-                            className="coexpression-select-profile"
-                        />
+                    >
+                        <span>Query Profile:</span>
```

Comment:
I'm more okay with using the word Query here, as displayed to the user. though in the code I think we should not have this concept, just stick with x and y

Reply:
I think x and y will be confusing if the api endpoint is again again subject and query profiles... @oplantalech how is this named in the api now?

Reply:
@pieterlukasse @adamabeshouse Now the API has `profileA` for what it was the `subjectProfile` (the profile of the queried genetic entity) and `profileB` for the `queryProfile`. Here (frontend), `profileA` is `profileX` and `profileB` is `profileY`, in accordance with the position of the profiles in the plots.

Reply:
I think then we should go with X and Y it makes the code a *lot* more readable